omnigan/generator.py

```diff
@@ -60,9 +61,55 @@ def get_gen(opts, latent_shape=None, verbose=0, no_init=False):
         init_gain=opts.gen.p.init_gain,
         verbose=verbose,
     )
+
+    if opts.gen.p.use_pretrained:
+        weights = torch.load(opts.gen.p.path_G_pretrained)
+        new_params = G.painter.state_dict().copy()
+        for elem in new_params:
+            if match_up_spade_v2(elem) in weights.keys() and not elem.startswith(
+                "conv_img"
+            ):
+                if new_params[elem].size() == weights[match_up_spade_v2(elem)].size():
+                    new_params[elem] = weights[match_up_spade_v2(elem)]
+            # If we only want to match the upsampling layers (that have the  right dimentsion)
+            #  if elem.startswith("up"):
+            #     print(elem)
+            #     if match_up_spade(elem)!= "":
+            #         if new_params[elem].size()==weights[match_up_spade(elem)].size():
+            #             new_params[elem] = weights[match_up_spade(elem)]
+        G.painter.load_state_dict(new_params)
     return G
 
 
+def match_name(name):
+    if name == "weight_bar":
+        return "weight_orig"
+    else:
+        return name
+
+
+def match_up_spade(s):
+    s_ = s.split(".")
+    if "module" in s_:
+        s_.remove("module")
+    if int(s_[1]) in range(3):
+        return (".").join(["up_" + str(int(s_[1]) + 1), *s_[2:-1], match_name(s_[-1])])
+    else:
+        return ""
+
+
+def match_up_spade_v2(s):
+    s_ = s.split(".")
```

Comment:
can you add a docstring with an example of why this is needed?

Reply:
is the "v2" for spade (if so, what does it mean) or for the `match_up_spade` function? (if so, do you need the other version?)

Reply:
Yes, i was trying out different ways of matching weights but I think I'll only keep one of them